decks/20210201-jest-mocking.md

```diff
@@ -0,0 +1,259 @@
+   
+         await act(async () => {
+            const renderedComponent = renderComponent(<User />);
+   
+            queryByText = renderedComponent.queryByText;
+         });
+   
+         await waitFor(() => {
+            expect(name()).toBeInTheDocument(); // Name is rendered
+         });
+      });
+   });
+```
+<div class="filename">src/components/Avatar.test.tsx</div>
+
+   ---
+
+4. test for output #2
+```tsx
+  import axios from 'axios';
+
+  const mockedAxios = axios as jest.Mocked<typeof axios>;
+
+   describe('Avatar', () => {
+      it('should render error if api call fails', async () => {
+         // overwrite /api/user call response
+         mockedAxios.request.mockResolvedValueOnce(
+           Promise.resolve({
+              data: {
+                 name: 'Wolverine',
+                 firstName: 'Logan',
+                 lastName: '',
+                 isXMen: true,
+              }
+           })
+         );
+         
+         /* ... */
+      });
+   });
+```
+<div class="filename">src/components/Avatar.test.tsx</div>
+
+   ---
+
+4. test for output #3
+```tsx
+  import axios from 'axios';
+
+  const mockedAxios = axios as jest.Mocked<typeof axios>;
+
+   describe('Avatar', () => {
+      it('should render error if api call fails', async () => {
+         // overwrite /api/user call response
+         mockedAxios.request.mockRejectedValueOnce(
+         	'User API error for test'
+         );
+      	
+         let queryByText: any;
+
+         const error = () => queryByText('error text');
+   
+         await act(async () => {
+            const renderedComponent = renderComponent(<User />);
+   
+            queryByText = renderedComponent.queryByText;
+         });
+
+         expect(error()).toBeInTheDocument();
+      });
+   });
+```
+<div class="filename">src/components/Avatar.test.tsx</div>
+
+   ---
+
+![Deadpool whaaat](https://media.giphy.com/media/jlduSIjwROKJlOzI2N/giphy.gif)
+
+---
+
+### Recap
+
+-<!-- .element: class="fragment" --> Mocking in Jest is easily achievable & will make your Dev-Life much easier
+
+-<!-- .element: class="fragment" --> By relying on static data / code you spend less time writing tests and can use the gained time to do other (nicer) things - like learning (~~fighting~~) Typescript 😂
+
+-<!-- .element: class="fragment" --> In my current Project I mocked the s**t out of Jest and gained most of my knowledge #bigfan
+
+---
+
+### Questions?
+
+---
+
+### Nomination
+
+![Next Talk](assets/20210201/next-talk.jpg)<!-- .element: class="fragment" -->
+
+---
+
+### Sources:
+
+* [Advanced React Component Mocks with Jest and React Testing Library](https://medium.com/@ericdcobb/advanced-react-component-mocks-with-jest-and-react-testing-library-f1ae8838400b)
+* [Jest Docs - Manual Mock](https://jestjs.io/docs/en/manual-mocks)
+* [Mocking Axios in Jest + Testing Async Functions](https://www.youtube.com/watch?v=9Yrd4aZkse8)
+* Atupri Repo ;)
```

Comment:
Eventually, you want to remove this line if NDA is involved

Reply:
true - created a new PR for that